### ISSUE TITLE
[Bug]: panic: aws_vpc_peering_connection: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by camlow325** [hashicorp/terraform-provider-aws#27822]
### Terraform Core Version

1.0.8

### AWS Provider Version

4.17.1

### Affected Resource(s)

* aws_vpc_peering_connection

### Expected Behavior

Apply with resource successful.

### Actual Behavior

Terraform crashed.

### Terraform Configuration Files

```terraform
resource "aws_vpc_peering_connection" "this" {
  auto_accept = true
  peer_vpc_id = ...
  vpc_id      = ...

  accepter {
    allow_remote_vpc_dns_resolution = false
  }

  requester {
    allow_remote_vpc_dns_resolution = false
  }

  tags = {}
}
```

### Steps to Reproduce

`terraform apply`. Note that we have only seen this crash once so far, with the same code being applied each time.

### Panic Output

```
Stack trace from the terraform-provider-aws_v4.17.1_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5d553a0]

goroutine 1109 [running]:
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/vpc_peering_connection.go:376
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.modifyVPCPeeringConnectionOptions.func1()
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/vpc_peering_connection.go:354 +0x125
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/resource/wait.go:27 +0x56
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/resource/state.go:110 +0x21f
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/resource/state.go:83 +0x1dd

Error: The terraform-provider-aws_v4.17.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by ewbankkit** [comment]
https://github.com/hashicorp/terraform-provider-aws/blob/e346fa12ef0d95733bb438fa2e7be12c9e286b59/internal/service/ec2/vpc_peering_connection.go#L375-L379

**Comment by matth-boise** [comment]
encountered a similar error from a newer version of the provider
```
Stack trace from the terraform-provider-aws_v4.51.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x880c440]

goroutine 1466 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.vpcPeeringConnectionOptionsEqual(0xc0041d7e00?, 0xc002cc20d8?)
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/vpc_peering_connection.go:382
github.com/hashicorp/terraform-provider-aws/internal/service/ec2.modifyVPCPeeringConnectionOptions.func1()
	github.com/hashicorp/terraform-provider-aws/internal/service/ec2/vpc_peering_connection.go:360 +0x125
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/resource/wait.go:27 +0x56
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/resource/state.go:110 +0x1ff
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/resource/state.go:83 +0x1d8
```

ok to leave this as a comment in this issue?  or better to create a new issue?

**Comment by ewbankkit** [comment]
https://github.com/hashicorp/terraform-provider-aws/blob/e3c30ab70c7c50f85296ac3d0cc7d23f488f5e02/internal/service/ec2/vpc_peering_connection.go#L381-L385

called from

https://github.com/hashicorp/terraform-provider-aws/blob/e3c30ab70c7c50f85296ac3d0cc7d23f488f5e02/internal/service/ec2/vpc_peering_connection.go#L359-L363

**Comment by jar-b** [comment]
Closed by #30966, merged to `main` via #31392
